source/obstacle/ObstacleGenerator.hx

```diff
@@ -7,73 +7,78 @@ import flixel.util.FlxTimer;
 import obstacle.Obstacle;
 
 @:generic
-class ObstacleGenerator<T:Obstacle> extends FlxBasic
-{
+class ObstacleGenerator<T:Obstacle> extends FlxBasic {
 	private var spawnTimer:FlxTimer = new FlxTimer();
 
 	public var obstacles(default, null):FlxTypedGroup<T>;
 	public var baseObstacleParameters:ObstacleParameters;
 	public var variation:ObstacleVariation;
 	public var spawnDelay(default, set):Float;
 
-	public function new(spawnDelay:Float, baseObstacleParameters:ObstacleParameters,
-			variation:ObstacleVariation, obstacles:FlxTypedGroup<T>)
-	{
+	public function new(spawnDelay:Float, baseObstacleParameters:ObstacleParameters, variation:ObstacleVariation, obstacles:FlxTypedGroup<T>) {
+		// call parent constructor
 		super();
 
+		// set to the obstacle objects
 		this.obstacles = obstacles;
+		// the base parameters used to create the obstacles
 		this.baseObstacleParameters = baseObstacleParameters;
+		// the variation that will be applied to the obstacles
 		this.variation = variation;
+		// the delay between obstacle spawning
 		this.spawnDelay = spawnDelay;
-
+		// start the spawn timer
+		// used to spawn in an obstacle
 		spawnTimer.start(spawnDelay, generateObstacle, 0);
 	}
 
-	public function generateObstacle(timer:FlxTimer)
-	{
+	public function generateObstacle(timer:FlxTimer) {
 		var recycled:Obstacle = obstacles.recycle();
 		recycled.init(variation.apply(baseObstacleParameters));
 	}
 
-	public function set_spawnDelay(spawnDelay:Float):Float
-	{
+	public function set_spawnDelay(spawnDelay:Float):Float {
 		this.spawnDelay = spawnDelay;
-		if (spawnTimer.active == true)
-		{
+		if (spawnTimer.active == true) {
 			spawnTimer.time = this.spawnDelay;
 		}
 		return this.spawnDelay;
 	}
 }
 
-class ObstacleVariation
-{
+class ObstacleVariation {
 	public var y:Float;
 	public var speed:Float;
 	public var width:Float;
 	public var height:Float;
 
-	public function new(y:Float = 0, speed:Float = 0, width:Float = 0, height:Float = 0)
-	{
+	public function new(y:Float = 0, speed:Float = 0, width:Float = 0, height:Float = 0) {
+		// constructor
```

Comment:
Adding comments for clarification can be good, but this is an example of one that I think ends up being pretty low-value. 

There's some cost to the long-term readability of code when comments are breaking things up (usually not super high, but it's there) so it's important that the ones that are added give us real value that isn't communicated well from the code alone. 

Since constructors are always named `new()` I think this is one of those low-value ones